deployワークフローの`sam build --use-container`が失敗する

## 再現方法

1. mainブランチにpushする
2. deployワークフローが`sam build --use-container`で失敗する

```
/home/runner/work/_temp/setup-sam-8Pr5tW/.venv/lib/python3.14/site-packages/samtranslator/compat.py:2: UserWarning: Core Pydantic V1 functionality isn't compatible with Python 3.14 or greater. from pydantic import v1 as pydantic

（中略）

RuntimeError: no validator found for <class 'pydantic.v1.fields.UndefinedType'>, see `arbitrary_types_allowed` in Config Error: Process completed with exit code 1.
```

## 期待する動作

- `sam build --use-container`が正常に終了してデプロイされる

## ToDo

- [ ] deployワークフローの`aws-actions/setup-sam@v2`に`use-installer: true`を設定

--- a/.github/workflows/deploy.yml
+++ b/.github/workflows/deploy.yml
@@ -20,7 +20,7 @@ jobs:
           use-installer: true
       - uses: aws-actions/configure-aws-credentials@v5
         with:
-          role-to-assume: arn:aws:iam::851307004996:role/Deploy-Workflow
+          role-to-assume: ${{ secrets.ASSUME_ROLE_ARN }}
           aws-region: ap-northeast-1
       - run: sam build --use-container
       - run: sam deploy --no-confirm-changeset --no-fail-on-empty-changeset

